### ISSUE TITLE
Examples Build Fails

### ISSUE THREAD

**Issue by scottfurry** [libgit2/libgit2#5049]
### Reproduction steps
on latest from master branch and happens always

### Actual behavior
Receive numerous errors during compile pointing to git_indexer_progress and git_indexer_options. Examples program lg2 fails to build. Error messages isolated to examples/clone.c, examples/fetch.c, and examples/index-pack.c.

Messages report similar to...
```
examples/clone.c:4:2: error: unknown type name ‘git_indexer_progress'
```

Suspect problem is in definition of type in include/git2/indexer.h where declaration is of the form
```
typedef struct git_indexer_progress {
...
} git_indexer_progress;
```
Same message is not reported when building main library or libgit2_clar.

### Version of libgit2 (release number or SHA1)
d1cfd79a477a440c5f0929c34e817da202227ec7

### Operating system(s) tested
Linux x86_64

Attached diff silences the errors but I am unsure if this is the correct fix for situation.

[indexer_diff.txt](https://github.com/libgit2/libgit2/files/3065042/indexer_diff.txt)

**Comment by ethomson** [comment]
How are you building the examples that this happens?  What compiler?  We build the examples in our CI builds on all platforms, with both gcc and clang and do not see this.

**Comment by scottfurry** [comment]
I knew I was forgetting something...
```
$ gcc --version
gcc (Funtoo 7.4.1-r6) 7.4.1 20181207
```
(Funtoo -> Gentoo variant)

I'm following similar to Gentoo ebuild (see [here - src_configure fn](https://gitweb.gentoo.org/repo/gentoo.git/tree/dev-libs/libgit2/libgit2-0.28.1.ebuild) ) albeit manually w/ cmake as follows:
```
$ cmake -H. -Bbuild -DUSE_SSH=TRUE -DTHREADSAFE=TRUE
```
Use of the cmake flags -H(src folder) -B(build or dest folder) are the only deviations from Gentoo build process.

**Comment by scottfurry** [comment]
I just realized my mistake...
I was trying to build examples "stand alone" when in fact they are callable from the main CMakeLists.txt in the project root directory with the `-DBUILD_EXAMPLES=True` flag. Doing that...everything worked.
tl;dr - bad command-line-fu. Apologies for the noise.
